### DISCUSSION TITLE
Support for DoT / DoH directly with Android 9+

### DISCUSSION THREAD

**Discussion by dot-mike** [chadgeary/cloudblock#77] [Ideas]
Android 9+ supports DoT with domain-name in the settings, DoH is supported as of Android 13 https://www.xda-developers.com/android-13-dns-https-support/

It would be really great to facilitate support for this directly so I don't have to use Wireguard!

**Comment by Dviros** [comment]
You don't want to open up your DNS server to the whole world :\

**Comment by dot-mike** [comment]
Good thought, but I don't have to... I can limit the IP-space allowed to connect :)

**Comment by chadgeary** [comment]
Hey - I've moved this to discussion for now. For cloudblock, the DoH service is provided by pointing pihole's standard dns traffic at a cloudflared container that encrypts the traffic and sends it to a DoH provider. I don't see Android's DoH support handling this scenario.
